lib/ProductOpener/Redis.pm

```diff
@@ -143,4 +145,130 @@ sub push_to_redis_stream ($user_id, $product_ref, $action, $comment, $diffs) {
 	return;
 }
 
+=head2 get_rate_limit_user_requests ($ip, $api_action)
+
+Return the number of requests performed by the given user for the current minute for the given API route.
+See https://redis.com/glossary/rate-limiting/ for more information.
+
+If the rate-limiter is not configured or if an error occurs, returns undef.
+
+=head3 Arguments
+
+=head4 String $ip
+The IP address of the user who is making the request.
+
+=head4 String $api_action
+The API action that is being requested.
+
+=cut
+
+sub get_rate_limit_user_requests ($ip, $api_action) {
+	if (!$redis_url) {
+		# No Redis URL provided, we can't get the remaining number of requests
+		if (!$sent_warning_about_missing_redis_url) {
+			$log->warn("Redis URL not provided, rate-limiting is disabled") if $log->is_warn();
+			$sent_warning_about_missing_redis_url = 1;
+		}
+		return;
+	}
+
+	my $error = "";
+	if (!defined $redis_client) {
+		# we were disconnected, try again
+		$log->info("Trying to reconnect to Redis");
+		init_redis();
+	}
+	my $resp;
+	if (defined $redis_client) {
+		$log->debug("Getting rate-limit remaining requests", {ip => $ip, api_action => $api_action})
+			if $log->is_debug();
+		my $current_minute = int(time() / 60);
+		eval {$resp = $redis_client->get("po-rate-limit:$ip:$api_action:$current_minute");};
```

Comment:
I would put key computation in a function, just to be sure we don't mess in a subtle way with it at some point (like in a refactor)...